### PR TITLE
Refactors queries to get user profile and addresses

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9695

## Description of Changes
- Make sure user profiles and addresses are shown in communities and threads
- Make sure you can search users

### CODE REVIEW

File: libs/schemas/src/queries/user.schemas.ts [hunk 1]
```diff
@@ -1,43 +1,84 @@
...
+      address: z.string(),
+      role: z.enum(Roles),
     }),
   ),
-  threads: z.array(Thread),
-  comments: z.array(Comment),
-  commentThreads: z.array(Thread),
-  isOwner: z.boolean(),
-  tags: z.array(Tags),
+  //group_ids: z.array(PG_INT),
```
Comment (masvelio) [inline libs/schemas/src/queries/user.schemas.ts:54]: what about it?

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx [hunk 2]
```diff
@@ -322,24 +322,23 @@ const useAuthentication = (props: UseAuthenticationProps) => {
...
-      const updatedProfiles =
-        await DISCOURAGED_NONREACTIVE_fetchProfilesByAddress(
-          // @ts-expect-error <StrictNullChecks>
-          account.profile.chain,
-          // @ts-expect-error <StrictNullChecks>
-          account.profile.address,
-        );
-      const currentUserUpdatedProfile = updatedProfiles[0];
-      if (!currentUserUpdatedProfile) {
+      const userAddresses = await trpcVanillaClient.user.getUserAddresses.query(
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:325]: can't you use trpc here instead of trpcVanillaClient ?

File: packages/commonwealth/client/scripts/utils/trpcClient.ts [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
-        };
+      headers() {
+        return buildHeaders();
       },
     }),
   ],
   transformer: undefined,
 });
+
+export const trpcVanillaClient = createTRPCProxyClient<API>({
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/utils/trpcClient.ts:33]: can you explain rationale here? I am not sure if this is a good approach to have multiple trpc clients? trpcClient is injected ino the react app in `App.tsx` and trpcVanillaClient is not. I assume that it might have some consequences eg in caching etc, however, I am not sure about it. What is the reason to introduce another client? cc Thoughts @mzparacha @burtonator ?

**Review by masvelio — COMMENTED** [review]
Got some errors when searching profiles.

https://github.com/user-attachments/assets/e85844fd-e723-44a9-a08d-d3bd7b3d2808



image

```
10:23:58.043] ERROR (utils.ts): INTERNAL_SERVER_ERROR: [SequelizeDatabaseError] user.searchUserProfiles: invalid reference to FROM-clause entry for table "Addresses" {"fingerprint":"INTERNAL_SERVER_ERROR-SequelizeDatabaseError-user.searchUserProfiles"}
    err: {
      "type": "DatabaseError",
      "message": "invalid reference to FROM-clause entry for table \"Addresses\"",
      "stack":
          Error
              at Query.run (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/dialects/postgres/query.js:76:25)
              at <anonymous> (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/sequelize.js:650:28)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at body (/Users/marcin/Desktop/Projects/commonwealth/libs/model/src/user/SearchUserProfiles.query.ts:56:24)
              at query (/Users/marcin/Desktop/Projects/commonwealth/libs/core/src/framework/query.ts:37:12)
              at <anonymous> (/Users/marcin/Desktop/Projects/commonwealth/libs/adapters/src/trpc/handlers.ts:134:24)
              at async resolveMiddleware (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:421:30)
              at async callRecursive (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:451:32)
              at async outputMiddleware (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:302:24)
              at async callRecursive (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:451:32)
      "name": "SequelizeDatabaseError",
      "parent": {
        "type": "DatabaseError",
        "message": "invalid reference to FROM-clause entry for table \"Addresses\"",
        "stack":
            error: invalid reference to FROM-clause entry for table "Addresses"
                at Parser.parseErrorMessage (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:369:69)
                at Parser.handlePacket (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:188:21)
                at Parser.parse (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:103:30)
                at Socket.<anonymous> (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/index.ts:7:48)
                at Socket.emit (node:events:518:28)
                at Socket.emit (node:domain:488:12)
                at addChunk (node:internal/streams/readable:559:12)
                at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
                at Readable.push (node:internal/streams/readable:390:5)
                at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
        "length": 187,
        "name": "error",
        "severity": "ERROR",
        "code": "42P01",
        "hint": "Perhaps you meant to reference the table alias \"a\".",
        "position": "583",
        "file": "parse_relation.c",
        "line": "3615",
        "routine": "errorMissingRTE",
        "sql": "SELECT\n          U.id AS user_id,\n          U.profile->>'name' AS profile_name,\n          U.profile->>'avatar_url' AS avatar_url,\n          U.created_at,\n          MAX(A.last_active) as last_active,\n          array_agg(\n            json_build_object(\n              'id', A.id,\n              'community_id', A.community_id,\n              'address', A.address,\n              'role', A.role\n            )\n          ) as addresses,\n          COUNT(U.id) OVER()::integer as total\n        FROM\n          \"Users\" U\n          JOIN \"Addresses\" A ON U.id = A.user_id \n        WHERE\n          \"Addresses\".community_id = 'dydx' AND\n          (U.profile->>'name' ILIKE '%peter%')\n        GROUP BY\n          U.id\n        ORDER BY last_active DESC\n        LIMIT 20\n        OFFSET 0"
      },
      "original": {
        "type": "DatabaseError",
        "message": "invalid reference to FROM-clause entry for table \"Addresses\"",
        "stack":
            error: invalid reference to FROM-clause entry for table "Addresses"
                at Parser.parseErrorMessage (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:369:69)
                at Parser.handlePacket (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:188:21)
                at Parser.parse (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:103:30)
                at Socket.<anonymous> (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/index.ts:7:48)
                at Socket.emit (node:events:518:28)
                at Socket.emit (node:domain:488:12)
                at addChunk (node:internal/streams/readable:559:12)
                at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
                at Readable.push (node:internal/streams/readable:390:5)
                at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
        "length": 187,
        "name": "error",
        "severity": "ERROR",
        "code": "42P01",
        "hint": "Perhaps you meant to reference the table alias \"a\".",
        "position": "583",
        "file": "parse_relation.c",
        "line": "3615",
        "routine": "errorMissingRTE",
        "sql": "SELECT\n          U.id AS user_id,\n          U.profile->>'name' AS profile_name,\n          U.profile->>'avatar_url' AS avatar_url,\n          U.created_at,\n          MAX(A.last_active) as last_active,\n          array_agg(\n            json_build_object(\n              'id', A.id,\n              'community_id', A.community_id,\n              'address', A.address,\n              'role', A.role\n            )\n          ) as addresses,\n          COUNT(U.id) OVER()::integer as total\n        FROM\n          \"Users\" U\n          JOIN \"Addresses\" A ON U.id = A.user_id \n        WHERE\n          \"Addresses\".community_id = 'dydx' AND\n          (U.profile->>'name' ILIKE '%peter%')\n        GROUP BY\n          U.id\n        ORDER BY last_active DESC\n        LIMIT 20\n        OFFSET 0"
      },
      "sql": "SELECT\n          U.id AS user_id,\n          U.profile->>'name' AS profile_name,\n          U.profile->>'avatar_url' AS avatar_url,\n          U.created_at,\n          MAX(A.last_active) as last_active,\n          array_agg(\n            json_build_object(\n              'id', A.id,\n              'community_id', A.community_id,\n              'address', A.address,\n              'role', A.role\n            )\n          ) as addresses,\n          COUNT(U.id) OVER()::integer as total\n        FROM\n          \"Users\" U\n          JOIN \"Addresses\" A ON U.id = A.user_id \n        WHERE\n          \"Addresses\".community_id = 'dydx' AND\n          (U.profile->>'name' ILIKE '%peter%')\n        GROUP BY\n          U.id\n        ORDER BY last_active DESC\n        LIMIT 20\n        OFFSET 0",
      "parameters": {}
    }
```


image
image

image

```
[10:22:38.780] ERROR (setupErrorHandlers.ts): syntax error at or near "" 
  AND ("" {"url":"/api/threads?community_id=dydx&search=peter&limit=20&page=1&order_by=rank&order_direction=DESC&include_count=true","method":"GET","headers":{"host":"localhost:3000","referer":"http://localhost:8080/dydx/search?q=peter&communityScope=dydx&sort=Best"},"user":{"id":128606},"status":500}
    err: {
      "type": "DatabaseError",
      "message": "syntax error at or near \"\" \n  AND (\"\"",
      "stack":
          Error
              at Query.run (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/dialects/postgres/query.js:76:25)
              at <anonymous> (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/sequelize@6.37.3_pg@8.11.5/node_modules/sequelize/src/sequelize.js:650:28)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at body (/Users/marcin/Desktop/Projects/commonwealth/libs/model/src/thread/SearchThreads.query.ts:89:23)
              at query (/Users/marcin/Desktop/Projects/commonwealth/libs/core/src/framework/query.ts:37:12)
              at getThreadsHandler (/Users/marcin/Desktop/Projects/commonwealth/packages/commonwealth/server/routes/threads/get_threads_handler.ts:165:27)
      "name": "SequelizeDatabaseError",
      "parent": {
        "type": "DatabaseError",
        "message": "syntax error at or near \"\" \n  AND (\"\"",
        "stack":
            error: syntax error at or near "" 
              AND (""
                at Parser.parseErrorMessage (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:369:69)
                at Parser.handlePacket (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:188:21)
                at Parser.parse (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:103:30)
                at Socket.<anonymous> (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/index.ts:7:48)
                at Socket.emit (node:events:518:28)
                at Socket.emit (node:domain:488:12)
                at addChunk (node:internal/streams/readable:559:12)
                at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
                at Readable.push (node:internal/streams/readable:390:5)
                at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
        "length": 101,
        "name": "error",
        "severity": "ERROR",
        "code": "42601",
        "position": "578",
        "file": "scan.l",
        "line": "1192",
        "routine": "scanner_yyerror",
        "sql": "SELECT \n  \"Threads\".id,\n  \"Threads\".title,\n  \"Threads\".body,\n  'thread' as type,\n  \"Addresses\".id as address_id,\n  \"Addresses\".user_id as address_user_id,\n  \"Addresses\".address,\n  \"Addresses\".community_id as address_community_id,\n  \"Threads\".created_at,\n  \"Threads\".community_id as community_id,\n  COUNT(*) OVER() AS total_count, \n  ts_rank_cd(\"Threads\".search, tsquery) as rank\nFROM \n  \"Threads\"\n  JOIN \"Addresses\" ON \"Threads\".address_id = \"Addresses\".id,\n  websearch_to_tsquery('english', $1) as tsquery\nWHERE\n  \"Threads\".deleted_at IS NULL\n  AND \"Threads\".community_id = $2\" \n  AND (\"Threads\".title ILIKE '%' || $1 || '%' \n  OR tsquery @@ \"Threads\".search)\nORDER BY rank DESC , \"Threads\".created_at DESC LIMIT $3 OFFSET $4",
        "parameters": [
          "peter",
          "dydx",
          20,
          0
        ]
      },
      "original": {
        "type": "DatabaseError",
        "message": "syntax error at or near \"\" \n  AND (\"\"",
        "stack":
            error: syntax error at or near "" 
              AND (""
                at Parser.parseErrorMessage (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:369:69)
                at Parser.handlePacket (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:188:21)
                at Parser.parse (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/parser.ts:103:30)
                at Socket.<anonymous> (/Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/pg-protocol@1.6.1/node_modules/pg-protocol/src/index.ts:7:48)
                at Socket.emit (node:events:518:28)
                at Socket.emit (node:domain:488:12)
                at addChunk (node:internal/streams/readable:559:12)
                at readableAddChunkPushByteMode (node:internal/streams/readable:510:3)
                at Readable.push (node:internal/streams/readable:390:5)
                at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
        "length": 101,
        "name": "error",
        "severity": "ERROR",
        "code": "42601",
        "position": "578",
        "file": "scan.l",
        "line": "1192",
        "routine": "scanner_yyerror",
        "sql": "SELECT \n  \"Threads\".id,\n  \"Threads\".title,\n  \"Threads\".body,\n  'thread' as type,\n  \"Addresses\".id as address_id,\n  \"Addresses\".user_id as address_user_id,\n  \"Addresses\".address,\n  \"Addresses\".community_id as address_community_id,\n  \"Threads\".created_at,\n  \"Threads\".community_id as community_id,\n  COUNT(*) OVER() AS total_count, \n  ts_rank_cd(\"Threads\".search, tsquery) as rank\nFROM \n  \"Threads\"\n  JOIN \"Addresses\" ON \"Threads\".address_id = \"Addresses\".id,\n  websearch_to_tsquery('english', $1) as tsquery\nWHERE\n  \"Threads\".deleted_at IS NULL\n  AND \"Threads\".community_id = $2\" \n  AND (\"Threads\".title ILIKE '%' || $1 || '%' \n  OR tsquery @@ \"Threads\".search)\nORDER BY rank DESC , \"Threads\".created_at DESC LIMIT $3 OFFSET $4",
        "parameters": [
          "peter",
          "dydx",
          20,
          0
        ]
      },
      "sql": "SELECT \n  \"Threads\".id,\n  \"Threads\".title,\n  \"Threads\".body,\n  'thread' as type,\n  \"Addresses\".id as address_id,\n  \"Addresses\".user_id as address_user_id,\n  \"Addresses\".address,\n  \"Addresses\".community_id as address_community_id,\n  \"Threads\".created_at,\n  \"Threads\".community_id as community_id,\n  COUNT(*) OVER() AS total_count, \n  ts_rank_cd(\"Threads\".search, tsquery) as rank\nFROM \n  \"Threads\"\n  JOIN \"Addresses\" ON \"Threads\".address_id = \"Addresses\".id,\n  websearch_to_tsquery('english', $1) as tsquery\nWHERE\n  \"Threads\".deleted_at IS NULL\n  AND \"Threads\".community_id = $2\" \n  AND (\"Threads\".title ILIKE '%' || $1 || '%' \n  OR tsquery @@ \"Threads\".search)\nORDER BY rank DESC , \"Threads\".created_at DESC LIMIT $3 OFFSET $4",
      "parameters": [
        "peter",
        "dydx",
        20,
        0
      ]
    }
    ```

File: libs/schemas/src/queries/user.schemas.ts (hunk 1 above)
Comment (Rotorsoft) [reply libs/schemas/src/queries/user.schemas.ts:54]: forgot to remove this, we don't use it anywhere

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx (hunk 2 above)
Comment (Rotorsoft) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:325]: I just replaced an implementation that was marked as TO_BE_DEPRECATED with the vanilla option b/c it was in a place where hooks are not possible according to the comments. If you think we can still use hooks here, we can change it to the react query client

File: packages/commonwealth/client/scripts/utils/trpcClient.ts (hunk 3 above)
Comment (Rotorsoft) [reply packages/commonwealth/client/scripts/utils/trpcClient.ts:33]: This is just a cleaner way to call the tRPC api, compared to the previous logic it replaces

**Comment by Rotorsoft** [comment]
> > @masvelio I cannot play the clip above, but I fixed something in the query. Can you pls try again and send me the URL that's failing if you can reproduce after the last commit? 🙇
> 
> other errors seems to be gone, now I am getting this on community that has contests
> 
> ```
> [16:07:54.776] ERROR (utils.ts): INTERNAL_SERVER_ERROR: [ZodError] contest.getAllContests: [
>   {
>     "code": "invalid_type",
>     "expected": "string",
>     "received": "number",
>     "path": [
>       1,
>       "contests",
>       0,
>       "score",
>       0,
>       "votes"
>     ],
>     "message": "Expected string, received number"
>   },
>   {
>     "code": "invalid_type",
>     "expected": "string",
>     "received": "number",
>     "path": [
>       3,
>       "contests",
>       0,
>       "score",
>       0,
>       "votes"
>     ],
>     "message": "Expected string, received number"
>   },
>   {
>     "code": "invalid_type",
>     "expected": "string",
>     "received": "number",
>     "path": [
>       4,
>       "contests",
>       0,
>       "score",
>       0,
>       "votes"
>     ],
>     "message": "Expected string, received number"
>   }
> ] {"fingerprint":"INTERNAL_SERVER_ERROR-ZodError-contest.getAllContests-invalid_type-invalid_type-invalid_type"}
>     err: {
>       "type": "ZodError",
>       "message": "[\n  {\n    \"code\": \"invalid_type\",\n    \"expected\": \"string\",\n    \"received\": \"number\",\n    \"path\": [\n      1,\n      \"contests\",\n      0,\n      \"score\",\n      0,\n      \"votes\"\n    ],\n    \"message\": \"Expected string, received number\"\n  },\n  {\n    \"code\": \"invalid_type\",\n    \"expected\": \"string\",\n    \"received\": \"number\",\n    \"path\": [\n      3,\n      \"contests\",\n      0,\n      \"score\",\n      0,\n      \"votes\"\n    ],\n    \"message\": \"Expected string, received number\"\n  },\n  {\n    \"code\": \"invalid_type\",\n    \"expected\": \"string\",\n    \"received\": \"number\",\n    \"path\": [\n      4,\n      \"contests\",\n      0,\n      \"score\",\n      0,\n      \"votes\"\n    ],\n    \"message\": \"Expected string, received number\"\n  }\n]",
>       "stack":
>           ZodError: [
>             {
>               "code": "invalid_type",
>               "expected": "string",
>               "received": "number",
>               "path": [
>                 1,
>                 "contests",
>                 0,
>                 "score",
>                 0,
>                 "votes"
>               ],
>               "message": "Expected string, received number"
>             },
>             {
>               "code": "invalid_type",
>               "expected": "string",
>               "received": "number",
>               "path": [
>                 3,
>                 "contests",
>                 0,
>                 "score",
>                 0,
>                 "votes"
>               ],
>               "message": "Expected string, received number"
>             },
>             {
>               "code": "invalid_type",
>               "expected": "string",
>               "received": "number",
>               "path": [
>                 4,
>                 "contests",
>                 0,
>                 "score",
>                 0,
>                 "votes"
>               ],
>               "message": "Expected string, received number"
>             }
>           ]
>               at get error [as error] (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/zod@3.23.6/node_modules/zod/lib/index.mjs:587:31)
>               at ZodArray.parseAsync (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/zod@3.23.6/node_modules/zod/lib/index.mjs:715:22)
>               at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>               at async outputMiddleware (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:308:26)
>               at async callRecursive (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:451:32)
>               at async callRecursive (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:451:32)
>               at <anonymous> (/Users/marcin/Desktop/Projects/commonwealth/libs/adapters/src/trpc/middleware.ts:180:22)
>               at async callRecursive (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:451:32)
>               at async callRecursive (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:451:32)
>               at async resolve (file:///Users/marcin/Desktop/Projects/commonwealth/node_modules/.pnpm/@trpc+server@10.45.2/node_modules/@trpc/server/dist/index.mjs:481:24)
>       "aggregateErrors": [
>         {
>           "type": "Object",
>           "message": "Expected string, received number",
>           "stack":
>               
>           "code": "invalid_type",
>           "expected": "string",
>           "received": "number",
>           "path": [
>             1,
>             "contests",
>             0,
>             "score",
>             0,
>             "votes"
>           ]
>         },
>         {
>           "type": "Object",
>           "message": "Expected string, received number",
>           "stack":
>               
>           "code": "invalid_type",
>           "expected": "string",
>           "received": "number",
>           "path": [
>             3,
>             "contests",
>             0,
>             "score",
>             0,
>             "votes"
>           ]
>         },
>         {
>           "type": "Object",
>           "message": "Expected string, received number",
>           "stack":
>               
>           "code": "invalid_type",
>           "expected": "string",
>           "received": "number",
>           "path": [
>             4,
>             "contests",
>             0,
>             "score",
>             0,
>             "votes"
>           ]
>         }
>       ],
>       "issues": [
>         {
>           "code": "invalid_type",
>           "expected": "string",
>           "received": "number",
>           "path": [
>             1,
>             "contests",
>             0,
>             "score",
>             0,
>             "votes"
>           ],
>           "message": "Expected string, received number"
>         },
>         {
>           "code": "invalid_type",
>           "expected": "string",
>           "received": "number",
>           "path": [
>             3,
>             "contests",
>             0,
>             "score",
>             0,
>             "votes"
>           ],
>           "message": "Expected string, received number"
>         },
>         {
>           "code": "invalid_type",
>           "expected": "string",
>           "received": "number",
>           "path": [
>             4,
>             "contests",
>             0,
>             "score",
>             0,
>             "votes"
>           ],
>           "message": "Expected string, received number"
>         }
>       ],
>       "name": "ZodError"
>     }
>     ```
> ```

Ok, should be fixed now... it was a change in contest schemas that I just merged from master that required a migration and adjustments to the view schema

**Review by masvelio — COMMENTED** [review]
Still getting zod error from contests 
Try this community http://localhost:8080/common/discussions

```
[09:59:27.023] ERROR (utils.ts): INTERNAL_SERVER_ERROR: [ZodError] contest.getAllContests: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      1,
      "contests",
      0,
      "score",
      0,
      "votes"
    ],
    "message": "Expected string, received number"
  },
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      3,
      "contests",
      0,
      "score",
      0,
      "votes"
    ],
    "message": "Expected string, received number"
  },
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "number",
    "path": [
      4,
      "contests",
      0,
      "score",
      0,
      "votes"
    ],
    "message": "Expected string, received number"
  }
]

```


Also getting different error when I want to mention user in comment or thread. Works fine on prod,

https://github.com/user-attachments/assets/455622c1-df63-4af4-a7ed-03aafb565584

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:325]: in this particular case we are in the hook so we can also use hook. Thus, I would prefer to use existing `useFetchProfilesByAddressesQuery`.

File: packages/commonwealth/client/scripts/utils/trpcClient.ts (hunk 3 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/utils/trpcClient.ts:33]: it's cleaner. The problem is that it is additional trpc client that lives in its different world. It has its own cache and it is not triggering react component rerenders. So for me there is no point to introduce it as from FE perspective it might create even more confusion.

File: packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx (hunk 2 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/useAuthentication.tsx:325]: Historical context: There are a of branching cases for auth here, and local react state updates caused issues with fetching from react query or using local state as a reference (notice the functions are using passed arguments instead of react-state for correct "state" of a var).

I suggest we use the vanilla client as using react-query/trpc would require us to update local state (with the id of profile) we want to fetch and then again listen/wait for this response, then use a `useEffect` for further logic.

In short: lets use the current implementation.

File: packages/commonwealth/client/scripts/utils/trpcClient.ts (hunk 3 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/utils/trpcClient.ts:33]: I agree with @masvelio. In practice we prefer and want to use react-query/trpc hooks instead of vanilla clients, but in some very rare cases we have to use vanilla client and for those I suggest we directly call the endpoint (caching can be easily traced) instead of introducing another query client which we'd to have to manage (separate caching).

I suggest we remove this and use an API call directly instead, we can typecase the response with the return type of the endpoint (just so we don't lose the DX benefit of trpc).

Comment (kurtisassad) [reply packages/commonwealth/client/scripts/utils/trpcClient.ts:33]: I think the rationale behind this new query client is that the deprecated call @Rotorsoft was removing (`DISCOURAGED_NONREACTIVE_fetchProfilesByAddress `) was not yet react-query. So instead of using the react-query trpc client and needing to refactor everything on the frontend, he had created a vanilla one just to replace this use case.

I also agree it should be in react-query, but this might bloat the PR too much. Maybe it is better to just keep it as is and create a ticket for frontend to migrate this query over to react query in the future.

**Review by masvelio — APPROVED** [review]

File: packages/commonwealth/client/scripts/state/api/profiles/fetchProfileById.ts [hunk 4]
```diff
@@ -1,113 +1,25 @@
...
-  return useQuery({
-    // eslint-disable-next-line @tanstack/query/exhaustive-deps
-    queryKey: [ApiEndpoints.FETCH_PROFILES_BY_ID, userId],
-    queryFn: () =>
-      fetchProfileById({
-        userId,
-        shouldFetchSelfProfile,
-      } as UseFetchProfileByIdQueryCommonProps),
-    // eslint-disable-next-line @tanstack/query/no-deprecated-options
-    onSuccess: (response) => {
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/state/api/profiles/fetchProfileById.ts:81]: removing this causes a bug with edit profile page

https://github.com/user-attachments/assets/f60977e6-c93d-4350-88d2-5593de184d0e

**Review by mzparacha — COMMENTED** [review]
Clicked around and tested different instances of user info/displays, worked well. Only found this 1 issue above. Otherwise LGTM. Thanks for all the hard work!

**Review by mzparacha — APPROVED** [review]

**Review by timolegros — APPROVED** [review]

File: packages/commonwealth/client/scripts/state/api/profiles/fetchProfileById.ts (hunk 4 above)
Comment (Rotorsoft) [reply packages/commonwealth/client/scripts/state/api/profiles/fetchProfileById.ts:81]: @mzparacha did you fix this? or just reporting the bug here? ... do you know how to fix it quickly if still a bug?

Comment (Rotorsoft) [reply packages/commonwealth/client/scripts/state/api/profiles/fetchProfileById.ts:81]: for more context, I removed it b/c it was creating an infinite loop in the client

Comment (mzparacha) [reply packages/commonwealth/client/scripts/state/api/profiles/fetchProfileById.ts:81]: > did you fix this?

nope, but going to give it a try locally.

Comment (mzparacha) [reply packages/commonwealth/client/scripts/state/api/profiles/fetchProfileById.ts:81]: added https://github.com/hicommonwealth/commonwealth/pull/9705/commits/59ad64b4bb4d254bc1985c722524e28dfcd85e96, which should fix this
